java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/cp/CPMapReleaseTest.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.simulator.tests.cp;
+
+import com.hazelcast.collection.IList;
+import com.hazelcast.collection.ISet;
+import com.hazelcast.cp.CPMap;
+import com.hazelcast.simulator.hz.HazelcastTest;
+import com.hazelcast.simulator.test.BaseThreadState;
+import com.hazelcast.simulator.test.annotations.AfterRun;
+import com.hazelcast.simulator.test.annotations.Prepare;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+import com.hazelcast.simulator.test.annotations.Verify;
+import com.hazelcast.simulator.tests.cp.helpers.CpMapOperationCounter;
+import com.hazelcast.simulator.utils.GeneratorUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertTrue;
+
+public class CPMapReleaseTest extends HazelcastTest {
+    // number of cp groups to host the created maps; if (distinctMaps % maps) == 0 then there's a uniform distribution of maps
+    // over cp groups, otherwise maps are allocated per-cp group in a RR-fashion. If cpGroups == 0 then all CPMap instances will
+    // be hosted by the default CPGroup. When cpGroups > 0, we create and host CPMaps across non-default CP Groups.
+    public int cpGroups = 0;
+    // number of distinct maps to create and use during the tests
+    public int maps = 1;
+    // number of distinct keys to create and use per-map; key domain is [0, keys)
+    public int keys = 1000;
+    // number of distinct value to create on every client (total number of values is valuesPerClient * clients)
+    public int valuesPerClient = 20;
+    // size in bytes for each key's associated value
+    public int valueSizeBytes = 100;
+    // set it to false if test should not check whether all values have been changed during the test run. Can be useful for short runs
+    public boolean checkInitialValueChanged = true;
```

Comment:
We can merge this test with the existing one I believe. We can add the `valuesPerClient` + `checkInitialValueChanged` and its associated logic. We can set defaults to `0` and respectively `false` for these (or keep as-is) and modify the logic accordingly as it looks like difference is minimal, otherwise we have two tests that are almost identical. WDYT?

Reply:
Basically it would me replace `CPMapTest` by this test + but use probs from original `CPMapTest`, right? I can do that, the main point why I wanted to have those tests separated is for the case when original `CPMapTest` should be updated somehow in the future for another performance testing etc. In that case we could want to keep `CPMapReleaseTest` in the original version.

Reply:
Duplication typically isn't good. So I would not mind if we could get rid of 1 of the tests.

Reply:
@pveentjer yup, they are already merged into `CPMapTest`.